My Project/clsGenerateTranslations.vb

```diff
@@ -0,0 +1,558 @@
+    ''' <returns>A datatable with 3 columns; form_name, control_name, id_text. </returns>
+    '''--------------------------------------------------------------------------------------------
+    Public Shared Function GetControlsDatatable(lstForms As List(Of Form)) As DataTable
+        Dim datatableControls As New DataTable
+        datatableControls.Columns.Add("form_name", GetType(String))
+        datatableControls.Columns.Add("control_name", GetType(String))
+        datatableControls.Columns.Add("id_text", GetType(String))
+
+        For Each frm As Form In lstForms
+            Dim dctComponents As Dictionary(Of String, Component) = New Dictionary(Of String, Component)
+            clsWinformsComponents.FillDctComponentsFromControl(frm, dctComponents)
+
+            For Each clsComponent In dctComponents
+                Dim idText As String
+
+                If TypeOf clsComponent.Value Is Control Then
+                    idText = GetActualTranslationText(DirectCast(clsComponent.Value, Control).Text)
+                ElseIf TypeOf clsComponent.Value Is ToolStripItem Then
+                    idText = GetActualTranslationText(DirectCast(clsComponent.Value, ToolStripItem).Text)
+                Else
+                    Throw New Exception("Developer Error: Translation dictionary entry (" & frm.Name & "," & clsComponent.Key & ") contained unexpected value type.")
+                    Exit For
+                End If
+                'add row of form_name, control_name, id_text
+                datatableControls.Rows.Add(frm.Name, clsComponent.Key, idText)
+            Next
+
+            'Special case for radio buttons in panels: 
+            '  Before the dialog is shown, each radio button is a direct child of the dialog 
+            '  (e.g. 'dlg_Augment_rdoNewDataframe'). After the dialog is shown, the raio button becomes 
+            '  a direct child of its parent panel.
+            '  Therefore, we need to show the dialog before we traverse the dialog's control hierarchy.
+            '  Unfortunately showing the dialog means that it has to be manually closed. So we only 
+            '  show the dialog for this special case to save the developer from having to manually 
+            '  close too many dialogs.
+            '  TODO: launch each dialog in a new thread to avoid need for manual close?
+            'If strTemp.ToLower().Contains("pnl") AndAlso strTemp.ToLower().Contains("rdo") Then
+            '    'frmTemp.ShowDialog()
+            '    frmTemp.Show()
+            '    strTemp = GetControls(frmTemp)
+            '    frmTemp.Close()
+            'End If
+        Next
+
+        Return datatableControls
+    End Function
+
+    'todo. can probably be improved futher to include "DoNotTranslate".
+    'also the heuristics here should be defined at the product level. see issue #4
+    Private Shared Function GetActualTranslationText(strText As String) As String
```

Comment:
Add function comment header

Reply:
I saw that you decided not to add a function header. It's a private function so this is OK. I resolved this comment.